src/libraries/System.Runtime/tests/System.IO.FileSystem.Tests/RandomAccess/NonSeekable.cs

```diff
@@ -0,0 +1,429 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO.Pipes;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Win32.SafeHandles;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    [SkipOnPlatform(TestPlatforms.Browser, "async file IO is not supported on browser")]
+    public class RandomAccess_NonSeekable : FileSystemTest
+    {
+        private const int VectorCount = 10;
+        private const int BufferSize = 3;
+        private const int VectorsByteCount = VectorCount * BufferSize;
+
+        protected virtual PipeOptions PipeOptions => PipeOptions.None;
+
+        private async Task<(SafeFileHandle readHandle, SafeFileHandle writeHandle)> GetNamedPipeHandlesAsync()
+        {
+            string name = GetNamedPipeServerStreamName();
+
+            var server = new NamedPipeServerStream(name, PipeDirection.In, -1, PipeTransmissionMode.Byte, PipeOptions);
+            var client = new NamedPipeClientStream(".", name, PipeDirection.Out, PipeOptions);
+
+            await Task.WhenAll(server.WaitForConnectionAsync(), client.ConnectAsync());
+
+            bool isAsync = (PipeOptions & PipeOptions.Asynchronous) != 0;
+            return (GetFileHandle(server, isAsync), GetFileHandle(client, isAsync));
```

Comment:
On Unix, the handles are set to non-blocking when they are used in an async operation.
Because these handles weren't used yet, the handles are non-blocking also when `PipeOptions.Asynchronous` is set in `RandomAccess_NonSeekable_AsyncHandles`.

If the handles were non-blocking, some of the reads/writes should fail with `EAGAIN`.

Reply:
When making the handles non-blocking by adding:
```cs
if (isAsync)
{
    await client.WriteAsync(new byte[1]);
    await server.ReadAsync(new byte[1]);
}
```
6 of the `RandomAccess_NonSeekable_AsyncHandles` tests fail with:
```
  Error Message:
   System.IO.IOException : The process cannot access the file because it is being used by another process.
```
This is the `EAGAIN` that is surfacing.

Reply:
> This is the EAGAIN that is surfacing.

Great catch! In such case, we should most likely do a bigger refactor and use the epoll code path here

Reply:
We had some discussions in the past of making the epoll path more generic and usable elsewhere.
So far we've avoided the effort and used it as-is (like in https://github.com/dotnet/runtime/pull/44647).